src/LanguageServer/Impl/PathsWatcher.cs

```diff
@@ -83,7 +84,7 @@ internal sealed class PathsWatcher : IDisposable {
 
         private void TimerProc(object o) {
             lock (_lock) {
-                if (_changedSinceLastTick) {
+                if (!_changedSinceLastTick) {
```

Comment:
I believe the old comparison was correct.
Otherwise it's not clear why later `_changedSinceLastTick` is set to `false`.

Reply:
It is throttling. Flag tells when there were no changes since last timer tick - i.e. multiple file change activity (such as installation of a large package) is over and it is time to reload modules. In fact, I am thinking 500 ms is too low.